src/librustc/middle/effect.rs

```diff
@@ -186,7 +186,7 @@ impl<'a, 'tcx, 'v> Visitor<'v> for EffectCheckVisitor<'a, 'tcx> {
             hir::ExprInlineAsm(..) => {
                 self.require_unsafe(expr.span, "use of inline assembly");
             }
-            hir::ExprPath(..) => {
+            hir::ExprPath(hir::QPath::Regular(..)) => {
```

Comment:
Nit: `hir::QPath::Regular(..)` is unnecessary, this commit contains several places where path kind is unnecessarily restricted, then `Def` is queried, they probably remained from previous version of this commit.

Nit 2: Unnecessary `..` -> `_` changes here and there.